### PR TITLE
Fix java client connection close

### PR DESCRIPTION
**Pull request by sunny19930321**
Find the connection of the Java client, always connect after the query, add the close method in the connection, close the connection when the detection PinotClientTransport has registered

### CODE REVIEW

File: pinot-api/src/main/java/org/apache/pinot/client/Connection.java [hunk 1]
```diff
@@ -156,4 +157,13 @@ public ResultSetGroup get(long timeout, TimeUnit unit)
       return new ResultSetGroup(response);
     }
   }
+
+  /**
+   * connection close.
+   */
+  public void close() {
+    if(Objects.nonNull(_transport)) {
+      System.exit(0);
```
Comment (kishoreg) [inline pinot-api/src/main/java/org/apache/pinot/client/Connection.java]: we should never have system.exit in the code

Comment (sunny209868) [reply pinot-api/src/main/java/org/apache/pinot/client/Connection.java]: Yes, So I put the deleted, in JsonAsyncHttpPinotClientTransport class increased the close method.

Comment (sunny209868) [reply pinot-api/src/main/java/org/apache/pinot/client/Connection.java]: If you do not add the close method, the client consistently maintains the connection state, it'll experience threads hanging and resource leaks，Thanks

**Comment by npawar** [comment]
Closed this PR due to 6 months inactivity. Reopen if needed.

**Pull request closed without merging**